### ISSUE TITLE
"sui start" panic (main branch 1.29)

### ISSUE THREAD

**Issue by mario4tier** [MystenLabs/sui#18468]
## Steps to Reproduce Issue

(1) Build sui binary from main branch (currently 1.29.0-f6f2584c5072 )

(2) `env RUST_LOG=error sui start --network.config /home/olet/suibase/workdirs/localnet/config/network.yaml`

network.yaml is attached ( [network.yaml.txt](https://github.com/user-attachments/files/16041178/network.yaml.txt) )

## Expected Result

"sui start" to normally start and keep running.

## Actual Result

A panic
![image](https://github.com/MystenLabs/sui/assets/4542410/4d15e1aa-6aab-49db-8b13-0d77c4f80bdb)

This problem is not happening with devnet branch 1.28.1 (and many older versions).

The error message shows a path that includes the file "network.yaml" which is not a directory (it is the file specified with --network-config), could this be the problem?

## System Information
ubuntu, rustc 1.75

Tests with 1.29 is my own cargo build, while 1.28.1 was the precompiled Mysten binaries ( sui-devnet-v1.28.1-ubuntu-x86_64.tgz )

**Comment by stefan-mysten** [comment]
Thanks for reporting this. I've made some changes to `sui start` that might be causing this. I'll look into it now, but would help me to know:
- where is your genesis blob stored (folder relative to localnet/config/network.yaml). I assume it is in `home/olet/suibase/workdirs/localnet`
- where are the previous DBs stored: are they in the `home/olet/suibase/workdirs/localnet`? Do you see a `full_node_db` there?

**Comment by mario4tier** [comment]
When testing, my script delete and regenerate "everything" from scratch. So in the case of testing 1.29, it does not reach the point of creating full_node_db.

(1) The blob is in /home/olet/suibase/workdirs/localnet/config

![image](https://github.com/MystenLabs/sui/assets/4542410/6f16fb2b-e7ff-4af5-9a5f-b378e687dbf5)

(2) Will follow up in a few minutes about where full_node_db is created when testing with 1.28.1

**Comment by stefan-mysten** [comment]
I see, in this case, I think you might just get away with passing the folder instead of the actual network file
`sui start --network.config ~/suibase/workdirs/localnet/config`.

**Comment by stefan-mysten** [comment]
But I will fix it asap so that it accepts the `network.yaml` config file as it used to.

**Comment by mario4tier** [comment]
Thanks. No rush to fix for me... as long it works as used to in devnet/testnet branch :smile: 

Just FYI, specifying just the directory did not work: 
![image](https://github.com/MystenLabs/sui/assets/4542410/4870990f-de17-4d58-a11d-66aedeb86ed4)

**Comment by mario4tier** [comment]
Going on a tangent about full_node_db (can create another issue as needed?)

With 1.28.1, it is created in `~/.sui/sui_config` ( it might have always work that way and never noticed).

This is inconsistent with all the rest being created in `~/suibase/workdirs/localnet/config`

![image](https://github.com/MystenLabs/sui/assets/4542410/7f3eaf0f-c841-4e1a-a70f-eb57e566ee4d)

**Comment by stefan-mysten** [comment]
Thanks for the detailed info! I'll try to put up a fix today.

**Comment by mario4tier** [comment]
More info related to full_node_db:
![image](https://github.com/MystenLabs/sui/assets/4542410/e16437c0-a1c6-4f71-a5fb-868018ffb63f)

**Comment by stefan-mysten** [comment]
thanks @mario4tier. I am not fully sure how `suibase` does stuff, particularly, why it creates another `config-default` folder to store the DBs. 

I have a PR up that should fix the issue you have. If you have a chance to build the `sui` binary from that branch https://github.com/MystenLabs/sui/pull/18469 and try it locally, let me know if you still encounter issues.

Thanks for your patience here!

**Comment by mario4tier** [comment]
It works (tested with latest 1.29 branch + changes to sui_command.rs)

About DBs location
==================
One key design choice for Suibase is to not interfere with the Mysten Labs official installation. So the Suibase localnet should not depend or even "touch" ~/.sui

That design allows to support a distinct set of binaries/repos/config/keystore for each network. The users/app can then operate localnet/devnet/testnet/mainnet concurrently... there is no "switch env" for Suibase users.

To keep things clean, all files related to a network are kept in a single place (e.g. ~/suibase/workdirs/testnet). full_node_db is the only exception right now...

I tried to modify the fullnode.yaml with an absolute path like done in network.yaml, but it did not work (still created in ~/.sui/sui_config ).

In short, ideally, Suibase would benefit from being able to specify the full_node_db path (but it is not breaking anything right now).
